code/modules/fishing/fishing.dm

```diff
@@ -0,0 +1,163 @@
+/*
+ * Fishing! Contains normal fishing rods and nets.
+ */
+
+#define FISHING_RARE     "rare"
+#define FISHING_UNCOMMON "uncommon"
+#define FISHING_COMMON   "common"
+#define FISHING_JUNK     "junk"
+#define FISHING_NOTHING  "nothing"
+
+/turf/simulated/floor/water
+	var/has_fish = TRUE //If the water has fish or not.
+
+	var/list/rare_fish_list = list(
```

Comment:
I highly suggest changing these to be GLOB lists, and have the water tiles point to that list in initialization instead of having each one have their own instances of lists, as that would be a waste of memory.